server/jetstream_cluster.go

```diff
@@ -4660,16 +4660,6 @@ func (js *jetStream) monitorConsumer(o *consumer, ca *consumerAssignment) {
 				js.setConsumerAssignmentRecovering(ca)
 			}
 
-			// Synchronize everyone to our state.
```

Comment:
Should we remove all together or adjust to make sure our upper state is correct?

It is there for a reason :)

Reply:
Do you remember what that reason was? I can't find anything written down as to why, the best I can find is from the commit message in cc77d66 which says:

> Also sync other consumers when taking over as leader but no need to process snapshots when we are in fact the leader.

As far as I can tell everything should work without this because the Raft machinery would be keeping the `ConsumerStore` up-to-date on the replicas in the usual way anyway, whereas sending snapshots from the upper layer directly opens up a potential for a race between the committed Raft node state and the applied upper layer state, but I'm not sure if I'm missing something here.

Reply:
We were experiencing state skew, and I wanted us synched. But could have been from a prior bug that has since been corrected.